sn_cli/src/subcommands/folders.rs

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::files::{download_file, upload_files, ChunkManager, UploadedFile, UPLOADED_FILES};
+
+use sn_client::{
+    Client, FilesApi, FolderEntry, FoldersApi, WalletClient, BATCH_SIZE, MAX_UPLOAD_RETRIES,
+};
+
+use sn_protocol::storage::{Chunk, ChunkAddress, RegisterAddress};
+use sn_transfers::HotWallet;
+
+use clap::Parser;
+use color_eyre::{
+    eyre::{bail, eyre},
+    Result,
+};
+use std::{
+    collections::BTreeMap,
+    fs::create_dir_all,
+    path::{Path, PathBuf},
+};
+use tokio::task::JoinSet;
+use walkdir::WalkDir;
+
+#[derive(Parser, Debug)]
+pub enum FoldersCmds {
+    Upload {
+        /// The location of the file(s) to upload for creating the folder on the network.
+        ///
+        /// Can be a file or a directory.
+        #[clap(name = "path", value_name = "PATH")]
+        path: PathBuf,
+        /// The batch_size to split chunks into parallel handling batches
+        /// during payment and upload processing.
+        #[clap(long, default_value_t = BATCH_SIZE, short='b')]
+        batch_size: usize,
+        /// Should the file be made accessible to all. (This is irreversible)
+        #[clap(long, name = "make_public", default_value = "false", short = 'p')]
+        make_public: bool,
+        /// The retry_count for retrying failed chunks
+        /// during payment and upload processing.
+        #[clap(long, default_value_t = MAX_UPLOAD_RETRIES, short = 'r')]
+        max_retries: usize,
```

Comment:
just a headup that this might be conflict/overlap with the PR https://github.com/maidsafe/safe_network/pull/1276/ ,
which using `RetryStrategy` instead